File: invoice/src/app/_model/store.ts
export class Store {
    constructor(
        public storeId?: string,
        public address?: string,
        public city?: string,
        public database?: string,
        public port?: number,
        public postCode?: string,
        public publicIp?: string,
        public storeName?: string,
        public serialNumber?: string,
        public macAddress?: string,
        public ServerName?: string,
        public hostedEnvironment?: string,
        public tick?: Date,
        public customerNo ?: string,
        public uri?: string,
        public dealerId: string = '',
        public customerId: string = ''
    ) {}
}

File: API/Models/Store.cs
using System.ComponentModel.DataAnnotations;

namespace API.Models
{
    public class Store
    {
        [Key]
        [StringLength(50)]
        public string StoreId { get; set; } = "";
        [StringLength(100)]
        public string Address { get; set; } = "";
        [StringLength(50)]
        public string City { get; set; } = "";
        [StringLength(50)]
        public string Database { get; set; } = "";
        public int Port { get; set; }
        [StringLength(10)]
        public string PostCode { get; set; } = "";
        [StringLength(50)]
        public string PublicIp { get; set; } = "";
        [StringLength(100)]
        public string StoreName { get; set; } = "";
        [StringLength(250)]
        public string SerialNumber { get; set; } = "";
        [StringLength(100)]
        public string MacAddress { get; set; } = "";
        public DateTime? Tick { get; set; }
        [StringLength(100)]
        public string Servername { get; set; } = "";
        [StringLength(100)]
        public string HostedEnvironment { get; set; } = "";
        [StringLength(20)]
        public string DealerId { get; set; } = "";
        [StringLength(20)]
        public string CustomerId { get; set; } = "";
        public bool IsBlocked { get; set; }
        public virtual ICollection<AssignedStores>? Users { get; set; }
    }
}